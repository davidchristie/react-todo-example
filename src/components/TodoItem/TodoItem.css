.TodoItem {
  box-shadow: 2px 2px 5px lightgray;  
  cursor: default;
  font-size: large;
  margin: 5px auto;
  padding: 10px;
  width: 80%;
}

.TodoItem.completed {
  color: lightgray;
  text-decoration: line-through;
}
