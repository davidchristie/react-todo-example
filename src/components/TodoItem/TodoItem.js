import PropTypes from 'prop-types'
import React from 'react'

import './TodoItem.css'

const TodoItem = props => {
  let className = 'TodoItem'
  if (props.item.completed) {
    className += ' completed'
  }
  return (
    <div className={className} onClick={props.onClick}>
      {props.item.text}
    </div>
  )
}

TodoItem.propTypes = {
  item: PropTypes.shape({
    completed: PropTypes.bool.isRequired,
    text: PropTypes.string.isRequired
  }).isRequired,
  onClick: PropTypes.func
}

export default TodoItem
