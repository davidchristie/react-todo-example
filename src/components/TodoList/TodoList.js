import PropTypes from 'prop-types'
import React from 'react'

import TodoItem from '../TodoItem'

const TodoList = props => {
  return (
    <div className='TodoList'>
      {
        props.items.map((item, index) => {
          return (
            <TodoItem
              item={item}
              key={index}
              onClick={() => props.toggleCompleted(index)}
            />
          )
        })
      }
    </div>
  )
}

TodoList.propTypes = {
  items: PropTypes.arrayOf(PropTypes.object).isRequired,
  toggleCompleted: PropTypes.func.isRequired
}

export default TodoList
