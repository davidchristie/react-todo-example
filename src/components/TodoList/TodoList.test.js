import { shallow } from 'enzyme'
import React from 'react'

import TodoItem from '../TodoItem'
import TodoList from './TodoList'

it('renders correctly', () => {
  const props = {
    items: [
      {
        completed: false,
        text: 'Test todo item 1'
      },
      {
        completed: true,
        text: 'Test todo item 2'
      },
      {
        completed: false,
        text: 'Test todo item 3'
      }
    ],
    toggleCompleted: () => {}
  }
  expect(shallow(<TodoList {...props} />)).toMatchSnapshot()
})

it('clicking on an item toggles completion', () => {
  const props = {
    items: [
      {
        completed: false,
        text: 'Test todo item 1'
      },
      {
        completed: true,
        text: 'Test todo item 2'
      },
      {
        completed: false,
        text: 'Test todo item 3'
      }
    ],
    toggleCompleted: jest.fn()
  }
  const itemIndex = 1
  const wrapper = shallow(<TodoList {...props} />)
  wrapper.find(TodoItem).at(itemIndex).simulate('click')
  expect(props.toggleCompleted.mock.calls[0]).toEqual([itemIndex])
})
