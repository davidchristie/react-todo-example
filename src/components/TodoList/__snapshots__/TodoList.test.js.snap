// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="TodoList"
>
  <TodoItem
    item={
      Object {
        "completed": false,
        "text": "Test todo item 1",
      }
    }
    key="0"
    onClick={[Function]}
  />
  <TodoItem
    item={
      Object {
        "completed": true,
        "text": "Test todo item 2",
      }
    }
    key="1"
    onClick={[Function]}
  />
  <TodoItem
    item={
      Object {
        "completed": false,
        "text": "Test todo item 3",
      }
    }
    key="2"
    onClick={[Function]}
  />
</div>
`;
